jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -13,25 +13,181 @@
 import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4jakarta.jdt.core.JDTUtils;
 import org.eclipse.lsp4jakarta.jdt.core.DiagnosticsCollector;
-import org.eclipse.lsp4jakarta.jdt.core.websocket.WebSocketConstants;
-import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.lsp4jakarta.jdt.core.JakartaCorePlugin;
+import org.eclipse.jdt.core.ISourceRange;
+import org.eclipse.lsp4jakarta.jdt.core.AnnotationUtil;
+import static org.apache.commons.lang3.ClassUtils.isPrimitiveOrWrapper;
+
+import static org.eclipse.lsp4jakarta.jdt.core.TypeHierarchyUtils.doesITypeHaveSuperType;
 
-import java.util.List;
 
 public class WebSocketDiagnosticsCollector implements DiagnosticsCollector {
-    public WebSocketDiagnosticsCollector() {
-    }
+	public WebSocketDiagnosticsCollector() {
+	}
+
+	private Diagnostic createDiagnostic(IJavaElement el, ICompilationUnit unit, String msg, String code) {
+		try {
+			ISourceRange nameRange = JDTUtils.getNameRange(el);
+			Range range = JDTUtils.toRange(unit, nameRange.getOffset(), nameRange.getLength());
+			Diagnostic diagnostic = new Diagnostic(range, msg);
+			diagnostic.setCode(code);
+			completeDiagnostic(diagnostic);
+			return diagnostic;
+		} catch (JavaModelException e) {
+			JakartaCorePlugin.logException(WebSocketConstants.DIAGNOSTIC_ERR_MSG, e);
+		}
+		return null;
+	}
+
+	@Override
+	public void completeDiagnostic(Diagnostic diagnostic) {
+		diagnostic.setSource(WebSocketConstants.DIAGNOSTIC_SOURCE);
+		diagnostic.setSeverity(WebSocketConstants.ERROR);
+	}
+
+	@Override
+	public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnostics) {
+		if (unit == null) {
+			return;
+		}
+
+		IType[] alltypes;
+		IField[] allFields;
+		IAnnotation[] allFieldAnnotations;
+		IMethod[] allMethods;
+		IAnnotation[] allMethodAnnotations;
+
+		HashMap<String, Boolean> checkWSEnd = null;
+
+		try {
+			alltypes = unit.getAllTypes();
+			for (IType type : alltypes) {
+				checkWSEnd = isWSEndpoint(type);
+
+				// checks if the class uses annotation to create a ws endpoint
+				if (checkWSEnd.get("isAnnotation")) {
+					invalidParamsCheck(type, WebSocketConstants.ON_OPEN, WebSocketConstants.ON_OPEN_PARAM_OPT_TYPES,
+							unit, diagnostics);
+				}
+
+			}
+		} catch (JavaModelException e) {
+			JakartaCorePlugin.logException(WebSocketConstants.DIAGNOSTIC_ERR_MSG, e);
+		}
+	}
+
+	private void invalidParamsCheck(IType type, String function, Set<String> validParamTypes, ICompilationUnit unit,
+			List<Diagnostic> diagnostics) throws JavaModelException {
+		// check methods annotations, then check their params, check that their type is
+		// any of the valid options, if it's a string, check that it has the @Params
+		// annotation
+		// If so, set up the severity to Error, and add a code?
+
+		IMethod[] allMethods = type.getMethods();
+
+		for (IMethod method : allMethods) {
+			IAnnotation[] allAnnotations = method.getAnnotations();
+
```

Comment:
On idea for getting the types on the method params: 

Example:
```java
@OnOpen
public void OnOpen(Integer boxIntParam, int intParam, String stringParam) throws IOException {
```

Iterating through the method.getParameters:
```java
for (ILocalVariable var : method.getParameters()) {
	String elemName = var.getElementName(); // resolves to boxIntParam, intParam, stringParam
	String typeSignature = var.getTypeSignature(); // resolves to QInteger, I, QString
}
```

Here is some info on type signatures: https://www.ibm.com/docs/en/rational-soft-arch/9.6.1?topic=SS8PJ7_9.6.1/org.eclipse.jdt.doc.isv/reference/api/org/eclipse/jdt/core/Signature.html 
Looks like Integer and String are "UnresolvedClassTypeSignatures"... so thinking we may need to hardcode check for these values.

Reply:
Improved on the above slightly:

```java

for (ILocalVariable var : method.getParameters()) {
	String elemName = var.getElementName(); // resolves to boxIntParam, intParam, stringParam
	String typeSignature = var.getTypeSignature(); // resolves to QInteger, I, QString
	String signature = var.getTypeSignature().replace("/", ".");
	 IType primaryType = var.getTypeRoot().findPrimaryType();
	String resolvedTypeName = JavaModelUtil.getResolvedTypeName(signature, primaryType); // resolved to java.lang.Integer, int, java.lang.String
	boolean test = true;

}

```

Reply:
@dalidia  This seems like the best way I can find so far, then we may still need to hardcode the accepted primitive and boxed primitive types and open a follow up issue to improve upon this. Let me know what you think
cc: @turkeylurkey @TrevCraw

Reply:
Thanks Kathryn! I ended up using that approach. Even though it works in manual testing, `JavaModelUtil.getResolvedTypeName(signature, primaryType)` fails to get the resolved type of a signature when writing the integration tests for all parameters type, returning a `null` instead. I debugged the tests and it successfully identifies the `primaryType`. However, it doesn't seem to be able to connect it to the signature. 

As an example of the current test behavior, when the integration tests run, the pluggin will always throw an error to the first parameter declared inside the method. This is because `resolvedTypeName` is [used to evaluate if it's a wrapper and if it's a Session](https://github.com/dalidia/lsp4jakarta/blob/onOpen-params-validation/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java#L116) In the example below, it will be session.
```java
@OnOpen
public void OnOpen(Session session, int intParam, String stringParam) throws IOException {
```

 @kathrynkodama  do you think the issue of not being able to get the resolved type is coming from the way the project was loaded for testing purposes? I was thinking that I could add a null checker for `JavaModelUtil.getResolvedTypeName(signature, primaryType)` and if it's null, either don't run diagnostics, or use the raw signature result (from `getTypeSignature`) . See example below:

```java
String signature = param.getTypeSignature();
String formatSignature = signature.replace("/", ".");
IType primaryType = param.getTypeRoot().findPrimaryType();
String resolvedTypeName = JavaModelUtil.getResolvedTypeName(formatSignature, primaryType);

if (resolvedTypeName == null) {
    // compare signature to "QString" || "QInteger" || ... instead of "java.lang.String" || "java.lang.Integer"
}
```

Reply:
This is interesting, I like the second suggestion of adding a null check and using the raw signature if it is null, we should open a follow up issue (labelled as a bug) for the test projects signature types not resolving correctly